docs/pages/management/guides/elastic-stack.mdx

```diff
@@ -0,0 +1,729 @@
+    # issued to the users with this role.
+    max_session_ttl: 10h
+
+  # This section declares a list of resource/verb combinations that are
+  # allowed for the users of this role. By default nothing is allowed.
+  allow:
+    impersonate:
+      users: ["teleport-event-handler"]
+      roles: ["teleport-event-handler"]
+```
+
+Next, create the role:
+
+```code
+$ tctl create teleport-event-handler-impersonator.yaml
+```
+
+Assign this role to your Teleport user by running the following commands:
+
+```code
+$ TELEPORT_USER=$(tsh status -f json | jq -r '.active.username')
+$ tctl get users/${TELEPORT_USER?} > out.yaml
+```
+
+Edit `out.yaml`, adding `teleport-event-handler-impersonator` to the list of
+existing roles.
+
+Apply your changes:
+
+```code
+$ tctl create -f out.yaml
+```
+
+Log out of your Teleport cluster and log in again to assume the new role.
+
+### Export the access plugin identity
+
+Like all Teleport users, `teleport-event-handler` needs signed credentials in
+order to connect to your Teleport cluster. You will use the `tctl auth sign`
+command to request these credentials for the plugin. 
+
+<ScopedBlock scope={["enterprise", "oss"]}>
+
+The format of the credentials depends on whether you have set up your network to
+give the plugin direct access to the Teleport Auth Service, or if all Teleport
+clients and services connect to the Teleport Proxy Service instead.
+
+<Tabs>
+<TabItem label="Connect to the Proxy Service">
+
+The following `tctl auth sign` command impersonates the `access-plugin` user,
+generates signed credentials, and writes an identity file to the local
+directory:
+
+```code
+$ tctl auth sign --user=teleport-event-handler --out=auth.pem
+```
+
+The Event Handler plugin listens for audit events by connecting to the Teleport
+Auth Service's gRPC endpoint over TLS.
+
+The identity file, `auth.pem`, includes both TLS and SSH credentials. Your
+Event Handler plugin uses the SSH credentials to connect to the Proxy Service,
+which establishes a reverse tunnel connection to the Auth Service. The plugin
+uses this reverse tunnel, along with your TLS credentials, to connect to the
+Auth Service's gRPC endpoint.
+
+You will refer to this file later when configuring the plugin.
+
+</TabItem>
+<TabItem label="Connect to the Auth Service">
+
+If your network allows your plugin to access the Auth Service directly, e.g.,
+you are running the plugin on the Auth Service host, the plugin uses TLS
+credentials to connect to the Auth Service's gRPC endpoint and listen for audit
+events.
+
+You can generate TLS credentials with the following command:
+
+```code
+$ tctl auth sign --format=tls --user=teleport-event-handler --out=auth
+```
+
+This command should result in three PEM-encoded files: `auth.crt`,
+`auth.key`, and `auth.cas` (certificate, private key, and CA certs
+respectively). Later, you will configure the plugin to use these credentials to
+connect to the Auth Service.
+
+</TabItem>
+</Tabs>
```

Comment:
This was as far as I was able to get while testing out this guide:

```
> tctl auth sign --user=teleport-event-handler --out=auth.pem
ERROR: access denied: impersonation is not allowed

> tctl auth sign --format=tls --user=teleport-event-handler --out=auth
ERROR: access denied: impersonation is not allowed
```

To debug I went back and ran the previous steps. What I discovered is that the new role assigned to the user doesn't seem to persist after logging out and back in:

```
> tctl create -f out.yaml
user "alex@fornuto.com" has been updated
> tctl users ls
User                   Roles                                                                    
---------------------- ------------------------------------------------------------------------ 
alex                   editor,access                                                            
alex@fornuto.com       editor,windows-desktop-admins,access,teleport-event-handler-impersonator 
teleport-event-handler teleport-event-handler                                                   

> tsh logout
Logged out all users from all proxies.
> tsh login --proxy=tp.mydomain.tld
If browser window does not open automatically, open it by clicking on the link:
 http://127.0.0.1:35095/4ddf90a0-b4b0-46d5-9a32-01974cd08073
> Profile URL:        https://tp.alexf.xyz:443
  Logged in as:       alex@fornuto.com
  Cluster:            mydomain.tld
  Roles:              access, editor, windows-desktop-admins
  Logins:             -teleport-nologin-0ca21c8b-3aee-4fdb-b9e0-b162d4e95d8c, -teleport-internal-join
  Kubernetes:         enabled
  Valid until:        2022-09-24 03:42:13 -0500 CDT [valid for 12h0m0s]
  Extensions:         permit-agent-forwarding, permit-port-forwarding, permit-pty

Teleport v10.2.2 is now available, please consider upgrading your cluster.

> tctl users ls
User                   Roles                                
---------------------- ------------------------------------ 
alex                   editor,access                        
alex@fornuto.com       editor,windows-desktop-admins,access 
teleport-event-handler teleport-event-handler       
```

(As an aside, `tctl` is still reporting that I need to upgrade but `tctl status` shows `Version  10.2.2`).

Not sure if this is a problem this doc should solve, or an additional step I need to take to allow impersonation, which is or should be documented elsewhere and linked in prerequisites.

Reply:
As we talked about internally, this looks like it's due to authenticating with an SSO user. I'll add a `Tabs` component with instructions for SSO vs. local users